stream_alert/rule_processor/threat_intel.py

```diff
@@ -411,11 +414,63 @@ def normalized_type_mapping(cls):
         return cls.__normalized_types
 
     @staticmethod
-    def is_public_ip(ip_address):
+    def valid_ip(ip_address):
+        """Verify that a ip_address string is valid
+
+        Args:
+            ip_address (string): address to be tested
+
+        Returns:
+            True if the ip_address is valid, otherwise False
+        """
+
         try:
-            ip_addr = IPAddress(str(ip_address))
-            # We also need to filter multicast ip which is a private ip. For example,
-            # multicast ip '239.192.0.1', is a private ip.
-            return ip_addr.is_unicast() and not ip_addr.is_private()
-        except: # pylint: disable=bare-except
+            IPAddress(ip_address)
+        except AddrFormatError:
+            LOGGER.error('IP address failed parse: %s', ip_address)
             return False
+        return True
+
+    @staticmethod
+    def in_network(ip_address, networks):
+        """Check that an ip_address is within a set of CIDRs
+
+        Args:
+            ip_address (str or netaddr.IPAddress): IP address to check
+            networks (set): IPNetwork instances
+
+        Returns:
+            Boolean representing if the given IP is within any CIDRs
+        """
+
+        if not StreamThreatIntel.valid_ip(ip_address):
+            return False
+
+        for cidr in networks:
+            if ip_address in cidr:
+                return True
+        return False
+
+    @staticmethod
+    def is_excluded_ioc(ioc_type, ioc_value):
+        """
+        check if we should bypass IOC lookup for specified IOC
+        Args:
+            ioc_type (string): the type of IOC to evaluate (md5, ip, domain)
+            value (string): the value of IOC to evaluate
+
+        Returns:
+            True if IOC lookup should be bypassed for this value
+            False if IOC should be looked up
+        """
+        if ioc_type == 'ip':
+            try:
+                ip_addr = IPAddress(str(ioc_value))
+                excluded_networks = StreamThreatIntel.excluded_iocs.get("ip", set())
+                return StreamThreatIntel.in_network(ip_addr, excluded_networks)
+            except: # pylint: disable=bare-except
+                LOGGER.error('IP IOC Exclusion Failed: %s', ioc_value)
+                return True
```

Comment:
We also need to handle `AddrFormatError` exception. In `cloudtrail_api` events, many of `source ipaddress` are `s3.amazonaws.com`, `ecs.amazonaws.com`, `cloudtrail.amazonaws.com`, etc. If we add such unit test cases will be super helpful.

Reply:
i believe this is already tested in `test_extract_ioc_from_record_with_private_ip` from #557 
i can extract it into its own test though if you think it should be its own test